hogben/tests/test_fisher.py

```diff
@@ -0,0 +1,354 @@
+    diagonal in the importance matrix:
+    g = [1.28125, 1.025, 76.875],
+        [0.5125, 0.41, 30.75],
+        [25.625, 20.5, 1537.5]
+    """
+    model = request.getfixturevalue(model_class)
+    xi = model.xi[:3]
+    for index, param in enumerate(xi):
+        param.importance = index + 1
+    mock_reflectivity.side_effect = generate_reflectivity_data()
+    g_correct = [
+        [1.28125, 1.025, 76.875],
+        [0.5125, 0.41, 30.75],
+        [25.625, 20.5, 1537.5]
+    ]
+    g_reference = fisher(QS, xi, COUNTS, [model])
+    np.testing.assert_allclose(g_reference, g_correct, rtol=1e-08)
+
+
+@pytest.mark.parametrize('model_class', ("refl1d_model",
+                                         "refnx_model"))
+@pytest.mark.parametrize('step', (0.01, 0.0075, 0.0025, 0.001, 0.0001))
+def test_fisher_consistent_steps(step, model_class, request):
+    """
+    Tests whether the Fisher information remains mostly consistent when
+    changing step size using the refnx model
+    """
+    model = request.getfixturevalue(model_class)
+    g_reference = fisher(QS, model.xi, COUNTS, [model], step=0.005)
+    g_compare = fisher(QS, model.xi, COUNTS, [model], step=step)
+    np.testing.assert_allclose(g_reference, g_compare, rtol=1e-02, atol=0)
+
+
+@patch('hogben.utils.reflectivity')
+@pytest.mark.parametrize('model_class', ("mock_refl1d_model",
+                                         "mock_refnx_model"))
+@pytest.mark.parametrize('model_params', (1, 2, 3, 4))
+def test_fisher_shape(mock_reflectivity, model_params, model_class, request):
+    """
+    Tests whether the shape of the Fisher information matrix remains
+     correct when changing the amount of parameters
+    """
+    model = request.getfixturevalue(model_class)
+    xi = model.xi[:model_params]
+
+    mock_reflectivity.side_effect = generate_reflectivity_data()
+
+    expected_shape = (model_params, model_params)
+    g = fisher(QS, xi, COUNTS, [model])
+    np.testing.assert_array_equal(g.shape, expected_shape)
+
+
+@patch('hogben.utils.reflectivity')
+@pytest.mark.parametrize('model_class', ("mock_refl1d_model",
+                                         "mock_refnx_model"))
+@pytest.mark.parametrize('qs',
+                         (np.arange(0.001, 1.0, 0.25),
+                          np.arange(0.001, 1.0, 0.10),
+                          np.arange(0.001, 1.0, 0.05),
+                          np.arange(0.001, 1.0, 0.01)))
+def test_fisher_diagonal_non_negative(mock_reflectivity, qs, model_class,
+                                   request):
+    """Tests whether the diagonal values in the Fisher information matrix
+     are all zero or greater"""
+    model = request.getfixturevalue(model_class)
+    mock_reflectivity.side_effect = (np.random.rand(len(qs)) for _ in range(9))
+    counts = [np.ones(len(qs)) * 100]
+    g = fisher([qs], model.xi, counts, [model])
+    assert min(np.diag(g)) >= 0 # Check if smallest value is at least zero
```

Comment:
I see what you are doing, but probably it is better to do:
`assert np.all(np.diag(g)) >= 0`